ngi_reports/stockholm/project_summary.py

```diff
@@ -161,64 +163,68 @@ def __init__(self, config, LOG, working_dir, **kwargs):
                 else:
                     self.LOG.info("PCR-free library was used, so setting fragment size as N/A")
                     self.samples_info[sample_id]['preps'][prep_id]['avg_size'] = "N/A"
-        
+
             if not self.samples_info[sample_id]['preps']:
                 self.LOG.warn('No library prep information was available for sample {}'.format(sample_id))
-        
+
         ## collect all the flowcell this project was run
-        self.get_project_flowcell(fcon.proj_list)
+        self.get_project_flowcell({'HiSeq2500':fcon.proj_list, 'HiSeqX':xcon.proj_list})
         if not self.flowcell_info:
             self.LOG.warn('There is no flowcell to process for project {}'.format(self.project_name))
             self.project_info['missing_fc'] = True
-            
-        ## Collect reuired information for all flowcell run for the project
+
+        ## different FC db have different key names, this will be changed when merging the dbs for X and 2500 FC
+        sample_qstat_keys = dict.fromkeys(['HiSeq2500','MiSeq'], {'qval':'% of >= Q30 Bases (PF)', 'bases':'# Reads'})
+        sample_qstat_keys['HiSeqX'] = {'qval':'% >= Q30bases', 'bases':'PF Clusters'}
+
+        ## Collect required information for all flowcell run for the project
         for fc in self.flowcell_info.values():
             fc_name = fc['name']
-            fc_obj = fcon.get_entry(fc['run_name'])
-            fc_runp = fc_obj.get('RunParameters',{})
+            fc_obj = xcon.get_entry(fc['run_name']) if fc['type'] == 'HiSeqX' else fcon.get_entry(fc['run_name'])
+            fc_runp = fc_obj.get('RunParameters',{}).get('Setup',{}) if fc['type'] == 'HiSeqX' else fc_obj.get('RunParameters',{})
             fc_illumina = fc_obj.get('illumina',{})
-            fc_run_summary = fc_illumina.get('run_summary',{})
-            self.flowcell_info[fc_name]['lanes'] = {}
-            
+            fc_lane_summary = fc_obj.get('lims_data', '').get('run_summary', {})
```

Comment:
This will raise an error if `lims_data` isn't present, as `get()` won't work on an empty string. I'm guessing you mean this:
```python
fc_lane_summary = fc_obj.get('lims_data', {}).get('run_summary', '')
```